### PR TITLE
Get rid of global instantiation of IPRoute and IPDB

### PR DESCRIPTION
**Pull request by fub2**

### CODE REVIEW

**Comment by fub2** [comment]
Run through functional test.

**Comment by weia5-emc** [comment]
Can line 247 be removed? it seems not to be used.

File: ivn/core.py [hunk 1]
```diff
@@ -512,19 +515,23 @@ def create_interface(self):
                 self.logger_topo.warning("ip link {} exists in namespace {} so not create it.".
                                          format(ifname, self.__namespace))
                 return
+        IP_ROUTE.close()
```
Comment (weia5-emc) [inline ivn/core.py]: better to move this line just before line 511.  "return" in 517 may jump out before calling IP_ROUTE.close()

File: ivn/core.py [hunk 2]
```diff
@@ -512,19 +515,23 @@ def create_interface(self):
...
             try:
                 veth.net_ns_fd = self.__namespace
             except netlink.exceptions.NetlinkError as e:
+                MAIN_IPDB.release()
                 if e.code == 17:  # "File exists"
                     pass
                 else:
                     raise e
 
+        MAIN_IPDB.release()
```
Comment (weia5-emc) [inline ivn/core.py:534]: used clause "finally" to combine line 528 and 534

**Review by weia5-emc — CHANGES_REQUESTED** [review]

**Review by weia5-emc — APPROVED** [review]

File: ivn/core.py (hunk 2 above)
Comment (fub2) [reply ivn/core.py:534]: There's failure reported in pyroute2 lib if we do this. Haven't figure out why it happens. Thanks for merging.

**Pull request merged**